command/agent/http.go

```diff
@@ -126,11 +126,12 @@ func (ln tcpKeepAliveListener) Accept() (c net.Conn, err error) {
 }
 
 // Shutdown is used to shutdown the HTTP server
-func (s *HTTPServer) Shutdown() {
+func (s *HTTPServer) Shutdown() error {
 	if s != nil {
 		s.logger.Printf("[DEBUG] http: Shutting down http server")
-		s.listener.Close()
+		return s.listener.Close()
```

Comment:
I still think we need to avoid returning an error from shutdown. One of the possible errors is closing a connection twice. If we hit that I think we'd end up in a situation where the HTTP server is shutdown and will continue to error on shutdown and not start without restarting the Nomad process.

https://play.golang.org/p/UbLMY6x6Js

I just don't know of anything useful we can do with a Close error.